src/main/java/jenkins/advancedqueue/jobinclusion/strategy/FolderBasedJobInclusionStrategy.java

```diff
@@ -62,19 +66,70 @@ public ListBoxModel getListFolderItems() {
 
 	};
 
+	static public class JobPattern {
+		private String jobPattern;
+
+		@DataBoundConstructor
+		public JobPattern(String jobPattern) {
+			this.jobPattern = jobPattern;
+		}
+
+	}
+
 	private String folderName;
 
+	private boolean useJobFilter = false;
+
+	private String jobPattern = ".*";
+
 	@DataBoundConstructor
-	public FolderBasedJobInclusionStrategy(String folderName) {
+	public FolderBasedJobInclusionStrategy(String folderName, JobPattern jobFilter) {
 		this.folderName = folderName;
+		this.useJobFilter = (jobFilter != null);
+		if (this.useJobFilter) {
+			this.jobPattern = jobFilter.jobPattern;
+		}
 	}
 
 	public String getFolderName() {
 		return folderName;
 	}
 
+	public boolean isUseJobFilter() {
+		return useJobFilter;
+	}
+
+	public String getJobPattern() {
+		return jobPattern;
+	}
+
 	@Override
 	public boolean contains(DecisionLogger decisionLogger, Job<?, ?> job) {
-		return job.getFullName().startsWith(folderName);
+		if (job.getFullName().startsWith(folderName)) {
+			if (!isUseJobFilter() || getJobPattern().trim().isEmpty()) {
+				decisionLogger.addDecisionLog(2, "Not using filter ...");
+				LOGGER.info("Not using filter ...");
+				return true;
+			} else {
+				decisionLogger.addDecisionLog(2, "Using filter ...");
+				LOGGER.info("Using filter ...");
```

Comment:
Such general-purpose logging provides few info to the admin, because it does not reference job. `INFO` level is also too high, maybe CONFIG or FINEST?